### PR TITLE
Fix : adds ignoreDuplicate flag to child models for bulkCreate function

### PR DESCRIPTION
**Pull request by viraj47**
## Pull Request Checklist

- [ ] Have you added new tests to prevent regressions? No
- [ ] If a documentation update is necessary, have you opened a PR to [the documentation repository] (https://github.com/sequelize/website/)?  No documentation update
- [ ] Did you update the typescript typings accordingly (if applicable)? Not required
- [ ] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?Yes
- [ ] Does the name of your PR follow [our conventions](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md#6-commit-your-modifications)?

## Description Of Change

bulkCreate function currently only applies ignoreDuplicate flag to parent model only. if child models are inlcuded in bulkCreate and it has duplicate values then Sequlize will throw error if there is unique constraint on the child table. The PR will add ignoreDuplicate flag to child table if there is a ignoreDuplicate flag on parent table. please refer to below issue.

https://github.com/sequelize/sequelize/issues/15583

## Todos

- [ ] 
- [ ] 
- [ ]

### CODE REVIEW

**Review by WikiRik — CHANGES_REQUESTED** [review]
Can you add a test for this? See https://github.com/sequelize/sequelize/issues/11605 as well which seems to be a duplicate of the issue you are fixing

**Comment by viraj47** [comment]
> Can you add a test for this? See #11605 as well which seems to be a duplicate of the issue you are fixing
 #11605 is the exact same issue. is there guideline to adding to adding tests ?

**Comment by WikiRik** [comment]
You can add an integration test under this if-block;
https://github.com/sequelize/sequelize/blob/5db695596dac4a2b366bb4793c3a2b7833f73e87/packages/core/test/integration/model/bulk-create.test.js#L432

Something like this should work, adapted from #11605 (does need some work on the expectations);
```
      it('should inherit the ignoreDuplicates options to included models', async function () {
        const data = [
          {
            username: 'John',
            Students: [{ no: 1, name: 'Sample Student Name' }],
          }, {
            username: 'Mike',
            Students: [{ no: 1, name: 'Another Student Name' }], // Student with already existing no
          },
        ];

        this.User.hasMany(this.Student);

        await this.customSequelize.sync({ force: true });

        await this.User.bulkCreate(data, { ignoreDuplicates: true, include: [this.Student] });

        const users = await this.User.findAll({ order: ['id'], include: [this.Student] });
        expect(users.length).to.equal(2);
        expect(users[0].Students.length).to.equal(1);
        expect(users[1].Students.length).to.equal(0);
      });
```

File: packages/core/src/model.js [hunk 1]
```diff
@@ -2334,7 +2334,7 @@ ${associationOwner._getAssociationDebugList()}`);
               logging: options.logging,
             })
```
Comment (WikiRik) [inline packages/core/src/model.js:2335]:
```suggestion
              logging: options.logging,
              ignoreDuplicates: options.ignoreDuplicates
            })
```

You can also do this instead
